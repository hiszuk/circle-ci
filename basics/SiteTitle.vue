<template>
  <div class="site-title">
    {{ title }}
  </div>
</template>

<script>
export default {
  name: 'SiteTitle',
  props: {
    title: {
      type: String,
      required: true
    }
  }
}
</script>
