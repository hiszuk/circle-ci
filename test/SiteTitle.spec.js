import { shallowMount } from '@vue/test-utils'
import SiteTitle from '@/basics/SiteTitle.vue'

describe('SiteTitle.vue', () => {
  const propsData = {
    title: 'Site Title'
  }

  describe('props', () => {
    test('propsの型が合っているか', () => {
      const wrapper = shallowMount(SiteTitle, { propsData })
      expect(typeof wrapper.vm.$props.title).toBe('string')
    })
  })

  describe('props', () => {
    test('propsが必須となっているか', () => {
      const wrapper = shallowMount(SiteTitle, { propsData })
      expect(wrapper.vm.$options.props.title.required).toBe(true)
    })
  })

  describe('props', () => {
    test('propsを受け取れること', () => {
      const wrapper = shallowMount(SiteTitle, { propsData })
      expect(wrapper.vm.$props.title).toBe(propsData.title)
    })
  })

  describe('template', () => {
    it('snapshot', () => {
      const wrapper = shallowMount(SiteTitle, { propsData })
      // レンダリング結果がスナップショットと同じか
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

})
