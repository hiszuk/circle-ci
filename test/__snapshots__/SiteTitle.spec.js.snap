// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteTitle.vue template snapshot 1`] = `
<div
  class="site-title"
>
  
  Site Title

</div>
`;
